src/injected/injectedScript.ts

```diff
@@ -228,111 +228,126 @@ export default class InjectedScript {
     }
     element.dispatchEvent(new Event('input', { 'bubbles': true }));
     element.dispatchEvent(new Event('change', { 'bubbles': true }));
-    return { value: options.filter(option => option.selected).map(option => option.value) };
+    return options.filter(option => option.selected).map(option => option.value);
   }
 
-  waitForEnabledAndFill(node: Node, value: string): types.InjectedScriptPoll<types.InjectedScriptResult<'notconnected' | 'needsinput' | 'done'>> {
+  waitForEnabledAndFill(node: Node, value: string): types.InjectedScriptPoll<'notelement' | 'notconnected' | 'notfillableinputtype' | 'notfillablenumberinput' | 'notfillableelement' | 'notvaliddate' | 'needsinput' | 'done'> {
     return this.poll('raf', progress => {
       if (node.nodeType !== Node.ELEMENT_NODE)
-        return { error: 'Node is not of type HTMLElement' };
-      const element = node as HTMLElement;
+        return 'notelement';
+      const element = node as Element;
       if (!element.isConnected)
-        return { value: 'notconnected' };
+        return 'notconnected';
       if (!this.isVisible(element)) {
         progress.logRepeating('    element is not visible - waiting...');
-        return false;
+        return this.continuePolling;
```

Comment:
Similarly to the previous review, I think that we should not drive the control flow with the return statements beyond the  simple filter/traversal patterns.

Can we re-craft `poll` to be like raf?

```js
const waitPredicate = progress => {
  if (!this.isVisible(element)) {
    progress.logRepeating('    element is not visible - waiting...');
    return this.requestAnimationFrame(waitPredicate);
  }
};

return this.requestAnimationFrame(waitPredicate);
```